src/releasers/dotnet.ts

```diff
@@ -0,0 +1,152 @@
+import {checkpoint, CheckpointType} from '../util/checkpoint';
+import {Update} from '../updaters/update';
+
+// Generic
+import {Changelog} from '../updaters/changelog';
+// Dotnet specific files
+import {CommonProperties} from '../updaters/dotnet/common-properties';
+import {Project} from '../updaters/dotnet/project';
+import {ReadMe} from '../updaters/dotnet/readme';
+
+export class DotNet extends ReleasePR {
+  static releaserName = 'dotnet';
+  // When release-please runs, it calls the method run() on the base
+  // ReleasePR class. run() performs operations common to all languages:
+  // checking if there is an unreleased release PR, checking if the release
+  // strategy supports snapshots.
+  //
+  // _run() is called from run(), with the expectation that it performs
+  // language specific release tasks, e.g., updating files specific to .NET.
+  protected async _run() {
+    if (this.snapshot) {
+      checkpoint('running as snapshot pre-release', CheckpointType.Success);
+    }
+    const latestTag = await this.gh.latestTag(
+      this.monorepoTags ? `${this.packageName}-` : undefined,
+      this.snapshot
+    );
+    const commits = await this.commits({
+      sha: latestTag ? latestTag.sha : undefined,
+      path: this.path,
+    });
+
+    const cc = new ConventionalCommits({
+      commits,
+      githubRepoUrl: this.repoUrl,
+      bumpMinorPreMajor: this.bumpMinorPreMajor,
+      changelogSections: this.changelogSections,
+    });
+    const candidate = await this.coerceReleaseCandidate(
+      cc,
+      latestTag,
+      this.snapshot
+    );
+
+    const changelogEntry = await cc.generateChangelogEntry({
+      version: candidate.version,
+      currentTag: `v${candidate.version}`,
+      previousTag: candidate.previousTag,
+    });
+
+    // don't create a release candidate until user facing changes
+    // (fix, feat, BREAKING CHANGE) have been made; a CHANGELOG that's
+    // one line is a good indicator that there were no interesting commits.
+    if (this.changelogEmpty(changelogEntry)) {
+      checkpoint(
+        `no user facing commits found since ${
+          latestTag ? latestTag.sha : 'beginning of time'
+        }`,
+        CheckpointType.Failure
+      );
+      return;
+    }
+
+    const updates: Update[] = [];
+
+    updates.push(
+      new Changelog({
+        path: 'docs/history.md',
+        changelogEntry,
+        version: candidate.version,
+        packageName: this.packageName,
+      })
+    );
+
+    updates.push(
+      new ReadMe({
+        path: 'README.md',
+        changelogEntry,
+        version: candidate.version,
+        packageName: this.packageName,
+      })
+    );
+
+    updates.push(
+      new CommonProperties({
+        path: 'src/CommonProperties.xml',
+        changelogEntry,
+        version: candidate.version,
+        packageName: this.packageName,
+      })
+    );
+
+    // Each project file has a reference to the version #. Update
```

Comment:
This is relatively rare - normally we'd just use project references. The Functions Framework is "odd" in this respect, because:

- We have project templates which need the version number
- We have examples which are deliberately pinned to a version to make Function deployment easier (well, *feasible* right now)